prime-router/src/main/kotlin/azure/BatchFunction.kt

```diff
@@ -51,8 +51,9 @@ class BatchFunction {
                     actionHistory.trackExistingInputReport(it.task.reportId)
                     report
                 }
-                val mergedReports = when (receiver.timing?.operation) {
-                    Receiver.BatchOperation.MERGE -> listOf(Report.merge(inReports))
+                val mergedReports = when {
+                    receiver.format == Report.Format.HL7 -> inReports  // don't merge, when we are about to split
+                    receiver.timing?.operation == Receiver.BatchOperation.MERGE -> listOf(Report.merge(inReports))
                     else -> inReports
```

Comment:
For HL7 Format, just below here we do a split(), which negates the merge().    Originally I was thinking this pointless merge was causing a bug.  Now, I don't think it was any more, but I left the code that avoids the pointless merge.